mesa/agent.py

```diff
@@ -50,19 +48,7 @@ def __init__(self, unique_id: int, model: Model) -> None:
         self.pos: Position | None = None
 
         # register agent
-        try:
-            self.model.agents_[type(self)][self] = None
-        except AttributeError:
-            # model super has not been called
-            self.model.agents_ = defaultdict(dict)
-            self.model.agents_[type(self)][self] = None
-            self.model.agentset_experimental_warning_given = False
-
-            warnings.warn(
-                "The Mesa Model class was not initialized. In the future, you need to explicitly initialize the Model by calling super().__init__() on initialization.",
-                FutureWarning,
-                stacklevel=2,
-            )
+        self.model.agents_[type(self)][self] = None
```

Comment:
What about `try-except` then raise error, instead of a warning, for now? I'm not sure if some people do read the release note for such a subtle change in the way they have to do `__init__()`.

Reply:
I don't think it's a bad idea, but with Mesa 3.0 I would like to clean the codebase and remove it altogether.

In the 3.0 release notes we can put a note to first update to the latest Mesa 2.x, which also contains this warning (and it now has been in there for all 2.2.x and 2.3.x releases).

Reply:
Thought about it some more, and I think you're actually right. It's better to throw an useful error than let it crash without any guidance.